File: C#PartTwo/Homework/Strings/CorrectBrackets/CorrectBrackets.cs
namespace CorrectBrackets
{
    using System;
    using System.Collections.Generic;

    class CorrectBrackets
    {
        static void Main(string[] args)
        {
            string input = Console.ReadLine();

            Stack<char> brackets = new Stack<char>();
            bool correct = true;
            for (int i = 0; i < input.Length; i++)
            {
                char currentSymbol = input[i];
                if (currentSymbol == '(')
                {
                    brackets.Push(currentSymbol);
                }
                else if (currentSymbol == ')' && brackets.Count == 0)
                {
                    correct = false;
                    break;
                }
                else if (currentSymbol == ')')
                {
                    brackets.Pop();
                }
            }

            if (correct && brackets.Count != 0 && brackets.Peek() == '(')
            {
                correct = false;
            }

            if (!correct)
            {
                Console.WriteLine("Incorrect");
            }
            else
            {
                Console.WriteLine("Correct");
            }

        }
    }
}

File: JavaScript-Fundamentals/Homeworks/11. Strings/02. Correct brackets/correct-brackets.js
function solve(args) {
    let text = args[0];

    let brackets = [];
    let correct = true;
    for (let i = 0; i < text.length; i+=1)
    {
        let currentSymbol = text[i];
        if (currentSymbol == '(')
        {
            brackets.push(currentSymbol);
        }
        else if (currentSymbol == ')' && brackets.length == 0)
        {
            correct = false;
            break;
        }
        else if (currentSymbol == ')')
        {
            brackets.pop();
        }
    }

    if (correct && brackets.length != 0 && brackets[brackets.length-1] == '(')
    {
        correct = false;
    }

    if (!correct)
    {
        console.log("Incorrect");
    }
    else
    {
        console.log("Correct");
    }
}